src/Pdf/index.ts

```diff
@@ -0,0 +1,254 @@
+import { Collection, CollectionsFilter, getCollections } from '../Collections';
+import { Bounds, isLocationInBounds, Location } from '../Geo';
+import { getPlaceDetailed, Place } from '../Places';
+import { getRoutesForTripDay } from '../Route';
+import { Day, getTripDetailed, ItineraryItem, Trip } from '../Trip';
+import { getStaticMap } from './DataAccess';
+import { PdfData, PdfDestination, PdfQuery, StaticMap, StaticMapSector} from './PdfData';
+
+export async function getPdfData(query: PdfQuery): Promise<PdfData> {
+	const trip: Trip = await getTripDetailed(query.tripId);
+	const pdfData: PdfData = {
+		destinations: [],
+		routes: []
+	};
+
+	if (!trip.days) {
+		return pdfData;
+	}
+
+	pdfData.destinations = await extractDestinationsFromTrip(trip);
+	pdfData.destinations = fillDestinationsWithPlacesFromCollections(
+		pdfData.destinations,
+		await getCollectionsForDestinations(pdfData.destinations)
+	);
+
+	pdfData.destinations = await generateDestinationsMaps(pdfData.destinations, query);
+
+	pdfData.routes = await Promise.all(trip.days.map((day: Day, dayIndex: number) => (
+		getRoutesForTripDay(trip.id, dayIndex)
+	)));
+
+	return pdfData;
+}
+
+export async function extractDestinationsFromTrip(trip: Trip): Promise<PdfDestination[]> {
+	interface PdfDestinationDataTmp {
+		destinationId: string;
+		mainMap: StaticMap|null;
+		secondaryMaps: StaticMap[];
+		places: Place[];
+	}
+
+	const destinations: PdfDestinationDataTmp[] = [];
+
+	trip.days!.forEach((day: Day) => {
+		day.itinerary.forEach(async (itineraryItem: ItineraryItem) => {
```

Comment:
- get destinations for trip using trip module method
- add method to places module/dao `getPlacesFromTrip` and use it to fetch places
- add places to destinations